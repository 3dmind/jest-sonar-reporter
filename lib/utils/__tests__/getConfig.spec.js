'use strict';

const getConfig = require('../getConfig');

describe('getConfig', () => {
  test('should retrieve config from JSON object', () => {
    expect.assertions(1);

    const json = {
      'jestSonar': {
        'reportPath': 'reports',
        'reportFile': 'test-report.xml',
        'indent': 4
      }
    };

    const config = getConfig(json);

    expect(config).toMatchSnapshot();
  });

  test('should default to empty config', () => {
    expect.assertions(1);

    const json = {};

    const config = getConfig(json);

    expect(config).toMatchSnapshot();
  });

  test('should include config for "env"', () => {
    expect.assertions(1);

    const json = {
      'jestSonar': {
        'reportPath': 'reports',
        'reportFile': 'test-report.xml',
        'indent': 4,
        'env': {
          'test': {
            'reportPath': 'reports-test'
          }
        }
      }
    };

    const config = getConfig(json, 'test');

    expect(config).toEqual({
      'reportPath': 'reports-test',
      'reportFile': 'test-report.xml',
      'indent': 4,
    });
  });
});
