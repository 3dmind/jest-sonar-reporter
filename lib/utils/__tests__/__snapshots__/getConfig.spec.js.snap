// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getConfig should default to empty config 1`] = `Object {}`;

exports[`getConfig should retrieve config from JSON object 1`] = `
Object {
  "indent": 4,
  "reportFile": "test-report.xml",
  "reportPath": "reports",
}
`;
